### PR TITLE
[WIP][TASK] Refactor extension

### PR DESCRIPTION
**Pull request by calien666**
This PR will take care of outdated usages and remove/replace old libraries with new ones.

Currently work in progress, not for use in production.

### CODE REVIEW

**Comment by franzholz** [comment]
Thanks for you contribution.

`$GLOBALS['TYPO3_REQUEST']->getParsedBody()` is called so many times.
Ths class Classes/Api/ActivityApi.php should have a global member variable with a reference to it.
E.g. `$requestBody`.

File: Classes/Api/BasketApi.php
Comment (franzholz) [inline Classes/Api/BasketApi.php:1]: I will fix this in an extra commit, because this is a bug.

**Comment by calien666** [comment]
> Thanks for you contribution.
> 
> `$GLOBALS['TYPO3_REQUEST']->getParsedBody()` is called so many times. Ths class Classes/Api/ActivityApi.php should have a global member variable with a reference to it. E.g. `$requestBody`.

Hi @franzholz, Please ignore at the moment, as I will refactor even more on this. Using `$GLOBALS['TYPO3_REQUEST']` should be avoided in favour of giving the $request to the method. This was a result of rector, refactoring old style `GU::_GP()`. There is a lots of work to do, I will continue on that.

**Comment by franzholz** [comment]
Change your PR into a draft:

https://docs.github.com/de/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request

**Comment by franzholz** [comment]
Most changes have been included.
In general no such big changes should be made in one PR. Try to build smaller PRs!

**Comment by calien666** [comment]
> Most changes have been included. In general no such big changes should be made in one PR. Try to build smaller PRs!

Not that big PRs is your opinion and in most cases you are right, but wn extension with this old and legacy built code can't be safely updated by small, just lightweight changes.

As you won't have others help, you won't get other's help. For myself, the work on this extension is done and I won't make any commitment here anymore.

This is the reason, I didn't continue work on this PR, as I tried getting this old busted code to work, but bricks between my feet held me up.

**Pull request closed without merging**